mm2src/mm2_main/src/lp_swap.rs

```diff
@@ -711,12 +712,25 @@ pub struct NegotiationDataV3 {
     taker_coin_htlc_pub: Vec<u8>,
 }
 
+#[derive(Clone, Debug, Eq, Deserialize, PartialEq, Serialize)]
+pub struct NegotiationDataV4 {
+    started_at: u64,
+    payment_locktime: u64,
+    secret_hash: Vec<u8>,
+    maker_coin_swap_contract: Vec<u8>,
+    taker_coin_swap_contract: Vec<u8>,
+    maker_coin_htlc_pub: Vec<u8>,
+    taker_coin_htlc_pub: Vec<u8>,
+    persistent_pubkey: Vec<u8>,
```

Comment:
Please try to avoid adding `NegotiationDataV4` message. I even think this new variant might break backwards compatibility.
I personally think it's better to store `maker_coin_htlc_pub` as `maker_pubkey` and `taker_coin_htlc_pub` as `taker_pubkey`. @shamardy what do you think?

Reply:
waiting for @shamardy feedback on this...

Reply:
I agree with @sergeyboyko0791 here. Using `persistent_pubkey` also leaks private information for private ARRR swaps.

Reply:
@borngraced is this PR ready for review? `NegotiationDataV4` is still not removed from the code.

Reply:
Oh sorry, I missed that! and yes it's ready for review.